src/frontend/Ast_to_Mir.ml

```diff
@@ -323,6 +323,18 @@ let rec param_size transform sizedtype =
   let k_choose_2 k =
     Expr.Helpers.(binop (binop k Times (binop k Minus (int 1))) Divide (int 2))
   in
+  let rec stoch_size f1 f2 st =
+    match st with
+    | SizedType.SMatrix (mem_pattern, d1, d2) ->
+        SizedType.SMatrix (mem_pattern, f1 d1, f2 d2)
+    | SArray (t, d) -> SizedType.SArray (stoch_size f1 f2 t, d)
+    | SInt | SReal | SComplex | SRowVector _ | SVector _ | STuple _
+     |SComplexRowVector _ | SComplexVector _ | SComplexMatrix _ ->
+        Common.ICE.internal_compiler_error
+          [%message "Expecting SMatrix, got " (st : Expr.Typed.t SizedType.t)]
+  in
+  let min_one d = Expr.Helpers.(binop d Minus (int 1)) in
+  let noop d = d in
```

Comment:
[optional] this function has a name -- `Fn.id` -- if you'd prefer to not define it yourself. Seems fine to leave, as this name may be clearer for someone not deeply familiar with the jane street libraries we use